core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -230,11 +243,29 @@ public void perform(Run<?,?> build, FilePath ws, Launcher launcher, TaskListener
         try {
             String artifacts = build.getEnvironment(listener).expand(this.artifacts);
 
-            Map<String,String> files = ws.act(new ListFiles(artifacts, excludes, defaultExcludes, caseSensitive));
+            FilePath dir = ws;
+            if (this.basePath != null && this.basePath.length() > 0) {
+                String path = build.getEnvironment(listener).expand(this.basePath);
```

Comment:
Pending [JENKINS-35671](https://issues.jenkins-ci.org/browse/JENKINS-35671), this should happen only if `build instanceof AbstractBuild`, and should also take into account `buildVariables`. (Same for `artifacts` but you are not touching that here.)